tests/jerry/es2015/regression-test-issue-2782.js

```diff
@@ -0,0 +1,22 @@
+// Copyright JS Foundation and other contributors, http://js.foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+array = [a];
```

Comment:
This test is a bit problematic imho, there are a lot of code element which does not relate to the problem. Do we need to have an undefined element in the array? the `a` variable is called in the loop and it is ignored via the try-catch. Do we need that function call/try-catch at all?

I would suggest reducing the PR to a minimal code which still triggers the issue in the old version.